src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -45,117 +73,128 @@ private sealed class Emitter
-            /// </summary>
-            private readonly HashSet<TypeMetadata> _typesWithMetadataGenerated = new();
+            private ContextGenerationSpec _currentContext = new();
 
-            /// <summary>
-            /// Types that were specified with System.Text.Json.Serialization.JsonSerializableAttribute.
-            /// </summary>
-            private readonly Dictionary<string, TypeMetadata> _rootSerializableTypes = null!;
+            private readonly SourceGenerationSpec _generationSpec = null!;
 
-            public Emitter(in GeneratorExecutionContext executionContext, Dictionary<string, TypeMetadata> rootSerializableTypes)
+            public Emitter(in GeneratorExecutionContext executionContext, SourceGenerationSpec generationSpec)
             {
                 _executionContext = executionContext;
-                _generationNamespace = $"{executionContext.Compilation.AssemblyName}.JsonSourceGeneration";
-                _rootSerializableTypes = rootSerializableTypes;
+                _generationSpec = generationSpec;
             }
 
             public void Emit()
             {
-                foreach (KeyValuePair<string, TypeMetadata> pair in _rootSerializableTypes)
+                foreach (ContextGenerationSpec contextGenerationSpec in _generationSpec.ContextGenerationSpecList)
                 {
-                    TypeMetadata typeMetadata = pair.Value;
-                    GenerateTypeMetadata(typeMetadata);
+                    _currentContext = contextGenerationSpec;
+
+                    foreach (TypeGenerationSpec typeGenerationSpec in _currentContext.RootSerializableTypes)
+                    {
+                        GenerateTypeInfo(typeGenerationSpec);
+                    }
+
+                    string contextName = _currentContext.ContextType.Name;
+
+                    // Add root context implementation.
+                    AddSource($"{contextName}.g.cs", GetRootJsonContextImplementation(), isRootContextDef: true);
+
+                    // Add GetJsonTypeInfo override implementation.
+                    AddSource($"{contextName}.GetJsonTypeInfo.g.cs", GetGetTypeInfoImplementation());
+
+                    // Add property name initialization.
+                    AddSource($"{contextName}.PropertyNames.g.cs", GetPropertyNameInitialization());
                 }
+            }
+
+            private void AddSource(string fileName, string source, bool isRootContextDef = false)
+            {
+                string? generatedCodeAttributeSource = isRootContextDef ? s_generatedCodeAttributeSource : null;
+
+                string wrappedSource = $@"// <auto-generated/>
 
-                // Add base default instance source.
-                _executionContext.AddSource("JsonContext.g.cs", SourceText.From(GetBaseJsonContextImplementation(), Encoding.UTF8));
+namespace {_currentContext.ContextType.Namespace}
```

Comment:
With the latest bits there is no longer a new line after the `{` in the declarations:

```C#
// <auto-generated/>

namespace AoTPlusJsonSG
{internal partial class JsonContext
    {private global::System.Text.Json.Serialization.Metadata.JsonTypeInfo<global::AoTPlusJsonSG.Person> _Person;
        public global::System.Text.Json.Serialization.Metadata.JsonTypeInfo<global::AoTPlusJsonSG.Person> Person
        {
            get
            {
```